### ISSUE TITLE
Discord vocabulary on Windows freezes entire UI when waiting for data

### ISSUE THREAD

**Issue by erg** [factor/factor#2788]
Looks like `read1` calls `SSL_read()` which is in blocking mode? This causes the ui to not respond but discord messages can still get through and process until it gets stuck at `read1` again.

We should set it to non-blocking and handle `SSL_ERROR_WANT_READ` and `SSL_ERROR_WANT_WRITE`

To set nonblocking, do this: `socket FIONBIO 1 set-ioctl-socket`

Not sure exactly where to change the code, maybe in `establish-connection`. `SSL_set_bio` `<ssl-socket>` are relevant

**Comment by mrjbq7** [comment]
this freezes the UI, but only on windows:

```factor
"5.161.56.241" 443 <inet4> ! factorcode.og
"factorcode.org" <secure>
binary [ read1 ] with-client
```

**Comment by mrjbq7** [comment]
Fixed as of ca2f2b869663fada432e4ef6a9b3e2745db6b843.
